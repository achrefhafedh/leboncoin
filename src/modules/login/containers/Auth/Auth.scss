@import '../../../../assets/colors.scss';

.log-form {
  width: 50%;
  min-width: 320px;
  max-width: 475px;
  background: $white;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.25);

  @media (max-width: 40em) {
    width: 95%;
    position: relative;
    margin: 2.5% auto 0 auto;
    left: 0%;
    transform: translate(0%, 0%);
  }

  .form {
    display: block;
    width: 100%;
    padding: 2em;
  }

  h2 {
    width: 100%;
    color: lighten($grey, 20%);
    font-family: 'open sans condensed';
    font-size: 1.35em;
    display: block;
    background: $grey;
    width: 100%;
    text-transform: uppercase;
    padding: 0.75em 1em 0.75em 1.5em;
    box-shadow: inset 0px 1px 1px fadeout($white, 95%);
    border: 1px solid darken($grey, 5%);
    text-shadow: 0px 1px 1px darken($grey, 5%);
    margin: 0;
    font-weight: 200;
  }

  input {
    display: block;
    margin: auto auto;
    width: 100%;
    margin-bottom: 2em;
    padding: 0.5em 0;
    border: none;
    border-bottom: 1px solid $light-grey;
    padding-bottom: 1.25em;
    font-size: 0.8rem;
    color: $medium-grey;
    &:focus {
      outline: none;
    }
  }

  .btn {
    display: inline-block;
    background: $blue;
    border: 1px solid darken($blue, 5%);
    padding: 1em 2em;
    color: $white;
    margin-right: 0.5em;
    font-size: 0.8rem;
    box-shadow: inset 0px 1px 0px fadeout($white, 80%);
    cursor: pointer;
    &:hover {
      background: lighten($blue, 5%);
    }
    &:active {
      background: $blue;
      box-shadow: inset 0px 1px 1px fadeout($black, 90%);
    }
    &:focus {
      outline: none;
    }
  }

  .guest {
    color: lighten($blue, 10%);
    line-height: 0.5em;
    position: relative;
    top: 2.5em;
    text-decoration: none;
    font-size: 0.75em;
    margin: 0;
    padding: 0;
    float: right;

    &:hover {
      color: darken($blue, 5%);
    }
  }
}
