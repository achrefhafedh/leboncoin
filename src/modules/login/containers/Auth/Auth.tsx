import React from 'react';
import './Auth.scss';

export const Auth = () => (
  <div className="log-form">
    <h2>Login to your account</h2>
    <div className="form">
      <input
        type="text"
        title="username"
        placeholder="username"
        value="admin"
      />
      <input
        type="password"
        title="username"
        placeholder="password"
        value="admin"
      />
      <button className="btn">Login</button>
      <a className="guest" href="#">
        Guest
      </a>
    </div>
  </div>
);
