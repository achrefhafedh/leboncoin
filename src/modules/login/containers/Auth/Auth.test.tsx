import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { AuthContainer } from './Auth';

configure({ adapter: new Adapter() });

describe('renders and execute function without crashing', () => {
  it('render without crashing', () => {
    const login = jest.fn();
    let mock: any = jest.fn();
    const auth = shallow(
      <AuthContainer
        match={mock}
        location={mock}
        history={mock}
        login={login}
      />
    );
    expect(auth).toMatchSnapshot();
  });

  it('execute function loginHandler', () => {
    const login = jest.fn();
    let mock: any = jest.fn();
    const historyMock: any = { push: jest.fn() };
    const user = {
      username: 'Achraf',
      password: '$#1Â°678Ac/+.',
    };
    const wrapper = shallow<AuthContainer>(
      <AuthContainer
        match={mock}
        location={mock}
        history={historyMock}
        login={login}
      />
    );
    wrapper.instance().loginHandler(user);
  });

  it('execute function guestLogged', () => {
    const login = jest.fn();
    let mock: any = jest.fn();
    const historyMock: any = { push: jest.fn() };
    const wrapper = shallow<AuthContainer>(
      <AuthContainer
        match={mock}
        location={mock}
        history={historyMock}
        login={login}
      />
    );
    wrapper.instance().guestLoggedHandler();
  });

  it('execute function changeValue', () => {
    const login = jest.fn();
    let mock: any = jest.fn();
    const event = {
      currentTarget: {
        value: 'value',
      },
    } as React.ChangeEvent<HTMLInputElement>;
    const wrapper = shallow<AuthContainer>(
      <AuthContainer
        match={mock}
        location={mock}
        history={mock}
        login={login}
      />
    );
    wrapper.instance().changeValueHandler(event, 'user');
  });
});
