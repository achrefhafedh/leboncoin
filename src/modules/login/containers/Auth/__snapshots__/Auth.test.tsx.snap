// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and execute function without crashing render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthContainer
    login={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Login to your account
        </h2>,
        <div
          className="form"
        >
          <input
            onChange={[Function]}
            placeholder="username"
            title="username"
            type="text"
          />
          <input
            onChange={[Function]}
            placeholder="password"
            title="username"
            type="password"
          />
          <button
            className="btn"
            onClick={[Function]}
          >
            Login
          </button>
          <div
            className="guest"
            onClick={[Function]}
          >
            Guest
          </div>
        </div>,
      ],
      "className": "log-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login to your account",
        },
        "ref": null,
        "rendered": "Login to your account",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="username"
              title="username"
              type="text"
            />,
            <input
              onChange={[Function]}
              placeholder="password"
              title="username"
              type="password"
            />,
            <button
              className="btn"
              onClick={[Function]}
            >
              Login
            </button>,
            <div
              className="guest"
              onClick={[Function]}
            >
              Guest
            </div>,
          ],
          "className": "form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "username",
              "title": "username",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "password",
              "title": "username",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Guest",
              "className": "guest",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Guest",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Login to your account
          </h2>,
          <div
            className="form"
          >
            <input
              onChange={[Function]}
              placeholder="username"
              title="username"
              type="text"
            />
            <input
              onChange={[Function]}
              placeholder="password"
              title="username"
              type="password"
            />
            <button
              className="btn"
              onClick={[Function]}
            >
              Login
            </button>
            <div
              className="guest"
              onClick={[Function]}
            >
              Guest
            </div>
          </div>,
        ],
        "className": "log-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login to your account",
          },
          "ref": null,
          "rendered": "Login to your account",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder="username"
                title="username"
                type="text"
              />,
              <input
                onChange={[Function]}
                placeholder="password"
                title="username"
                type="password"
              />,
              <button
                className="btn"
                onClick={[Function]}
              >
                Login
              </button>,
              <div
                className="guest"
                onClick={[Function]}
              >
                Guest
              </div>,
            ],
            "className": "form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "username",
                "title": "username",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "password",
                "title": "username",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Guest",
                "className": "guest",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Guest",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
